src/App.tsx

```diff
@@ -1,62 +1,87 @@
-import React from 'react';
+import React, {
+  useState,
+  useEffect,
+  useCallback,
+} from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
-import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
-import { Loader } from './components/Loader';
+import { getPostsByUserId } from './api/post';
+import { Content } from './components/Content';
+import { Sidebar } from './components/Sidebar';
+import { Post } from './types/Post';
+import { Error } from './types/Error';
 
 export const App: React.FC = () => {
-  return (
-    <main className="section">
-      <div className="container">
-        <div className="tile is-ancestor">
-          <div className="tile is-parent">
-            <div className="tile is-child box is-success">
-              <div className="block">
-                <UserSelector />
-              </div>
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [postId, setPostId] = useState<number | null>(null);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleError = useCallback((errorType: Error | null) => {
+    setError(errorType);
+  }, []);
+
+  const handleOnPost = useCallback((id: number | null) => {
```

Comment:
The 'handleOnPost' function name can be misleading. Consider renaming it to 'handlePostSelection' to better reflect its purpose.